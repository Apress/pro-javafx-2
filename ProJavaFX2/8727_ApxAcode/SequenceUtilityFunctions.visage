/*
 * @test
 * @run
 */

import visage.util.Sequences.*;

var seq = [1, 4, 2, 8, 5, 7];
print("seq = "); println(seq);

println("The index of 4 in seq = {indexOf(seq, 4)}");
println("The max value of seq = {max(seq)}");
println("The min value of seq = {min(seq)}");

print("reverse(seq) = "); println(reverse(seq));
print("shuffle(seq) = "); println(shuffle(seq));

var sorted = sort(seq);
print("sortd = "); println(sorted);

var index = binarySearch(sorted, 4);
println("Found 4 in sorted at index {index}");

var integers = [1, 3, 5, 3, 1];
print("integers = "); println(integers);
println("indexOf(integers, 3) = {indexOf(integers, 3)}");
println("nextIndexOf(integers, 3, 2) = {nextIndexOf(integers, 3, 2)}");
