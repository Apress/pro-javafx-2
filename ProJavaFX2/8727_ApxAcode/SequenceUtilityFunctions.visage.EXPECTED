seq = [ 1, 4, 2, 8, 5, 7 ]
The index of 4 in seq = 1
The max value of seq = 8
The min value of seq = 1
reverse(seq) = [ 7, 5, 8, 2, 4, 1 ]
// @Skip shuffle(seq) = [ 8, 2, 5, 4, 7, 1 ]
sortd = [ 1, 2, 4, 5, 7, 8 ]
Found 4 in sorted at index 2
integers = [ 1, 3, 5, 3, 1 ]
indexOf(integers, 3) = 1
nextIndexOf(integers, 3, 2) = 3
